Cargo.toml

```diff
@@ -11,6 +11,7 @@ license = "Apache-2.0"
 actix-web = { version = "4.4.0" }
 actix-web-httpauth = { version = "0.8.1" }
 async-trait = { version = "0.1.74" }
+ayb-hosted-db-runner = { path = "src/hosted_db_runner" }
```

Comment:
Maybe creating a [workspace](https://doc.rust-lang.org/book/ch14-03-cargo-workspaces.html) is a good idea.
There could be three packages: `ayb`, `ayb-hosted-db-runner`, and `ayb-common`.
That way things like AybError can be imported from the `ayb-common` crate. :)

Amazing work!! :)

Reply:
Thank you! This is a great recommendation. I'm trying to figure out the right structure for what we need. All we really want is a second binary crate for the runner so that we can wrap it in a `vsjail`. Would it be enough to create two binary crates (`ayb` and `hosted_db_runner`) in the same package and avoiding multiple packages or workspaces entirely, or can we think of some benefit of breaking them apart?

Relatedly, I'm curious if there's a canonical way to reference one binary crate from another so the `ayb` binary can say "run the other binary for the runner" without having to explicitly provide a path to it.

Reply:
> Would it be enough to create two binary crates (ayb and hosted_db_runner) in the same package

I have never seen that in the wild tbh. Maybe it's fine, but that's also one of the use cases of workspaces... I would create a top level workspace just to be sure and to also follow the "convention".

This article may be of interest: https://matklad.github.io/2021/08/22/large-rust-workspaces.html

> Relatedly, I'm curious if there's a canonical way to reference one binary crate from another

Not that I am aware of, sadly.
You could embed the runner at compile time with `include_bytes!` and create a copy of the runner at startup time and delete it when the program is stopped, that way the runner is always up-to-date and is also directly tied to the ayb binary.

Reply:
Phew! I managed to figure out a way without too much hackery and one somewhat safe assumption
* One crate
* Two binaries (`ayb` and ayb_isolated_runner`)
* Since it's the same crate, it's relatively safe to assume both binaries will end up in the same directory (true of build targets as well as where the files will be installed with a `crate install`
* If we assume they are in the same directory, we can call `std::env::current_exe` to get a path to the current binary, which we can then use to create a path to the other binary

Hooray! I'm resolving conflicts, but this PR is largely good for review if you want to take a pass over it. I've self-reviewed and left myself comments that I'll resolve in the coming days, but if you're interested I'd love your opinion! :)

Reply:
That sounds like some reasonable assumptions and a clean workaround :)
I still don't really like the idea of one crate having two binaries like that... maybe creating a `[[bin]]` entry is a better approach (https://doc.rust-lang.org/cargo/reference/cargo-targets.html#binaries)

I'll review the PR :)